Documentation/Storage-Configuration/Ceph-CSI/ceph-csi-drivers.md

```diff
@@ -238,3 +238,40 @@ parameters:
 ```
 
 * PVCs created using the new storageclass will be encrypted.
+
+
+### Enable Read affinity for RBD volumes
+
+Ceph CSI supports mapping RBD volumes with krbd options
+`"read_from_replica=localize,crush_location=type1:value1|type2:value2"` to
+allow serving reads from the most local OSD (according to OSD locations as
+defined in the CRUSH map).
+Refer [krbd-options](https://docs.ceph.com/en/latest/man/8/rbd/#kernel-rbd-krbd-options)
+for more details.
+
+Execute the following steps:
+
+* Add labels to Kubernetes nodes following procedure similar to how it is mentioned in
+[osd-topology](../../CRDs/Cluster/ceph-cluster-crd.md#osd-topology).
+
+* Update the `rook-ceph-operator-config` configmap and patch it using the following
+command.
+```console
+kubectl patch cm rook-ceph-operator-config -nrook-ceph -p $'data:\n "CSI_ENABLE_READ_AFFINITY": "true"'
+```
+
+* (optional) Rook will pass the labels mentioned in [osd-topology](../../CRDs/Cluster/ceph-cluster-crd.md#osd-topology)
+as the default set of labels. This can overridden to supply custom labels using the following command.
+```console
+kubectl patch cm rook-ceph-operator-config -nrook-ceph -p $'data:\n "CSI_CRUSH_LOCATION_LABELS": "<custom-labels>"'
+```
+
+Ceph CSI will extract the crush location and pass it as
+`"--options read_from_replica=localize,crush_location=zone:east-zone1|region:east"`
+krbd option while mapping RBD volumes.
+If enabled, this option will be added to all RBD volumes mapped by Ceph CSI.
+
+!!! note
+    Make sure the version of ceph-csi used is `v3.8.0+`.
```

Comment:
The 3.8 driver will come with Rook 1.11 where this feature would be enabled. I don't think we need to mention this version here.
```suggestion
```